### ISSUE TITLE
Control Library Styles Not Working

### ISSUE THREAD

**Issue by Soap-141** [unoplatform/uno#11436]
### Current behavior

Using this [documentation](https://platform.uno/docs/articles/guides/how-to-create-control-libraries.html) to create a control library for my existing Uno project doesn't work. The control never have it's default style used.

### Expected behavior

The control should have it's default style used.

### How to reproduce it (as minimally and precisely as possible)

1. Download [BringTemplate.zip](https://github.com/unoplatform/uno/files/10814978/BringTemplate.zip)
2. Start the solution with Visual Studio
3. Start the application on Windows
4. Notice that the page is empty

### Workaround

Using `Uno.XamlMerge.Task` and directly referencing project together instead of using a NuGet worked but that's not an option for the project.

### Works on UWP/WinUI

No

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

4.7.37

### Affected platforms

Android, iOS, Windows

### IDE

Visual Studio 2022

### IDE version

17.4.5

### Anything else we need to know?

I did not try the mobile platforms on the sample application because I was having weird issues with the new Uno application template (I did not compile and was complaining about targets). But I know from my other project that it also doesn't work on mobile.

**Comment by jeromelaban** [comment]
Thanks for the report.

There are multiple issues in your sample coming from misdirections in the documentation:
- "Custom Control" does not exist in the template. It's `Templated Control` that needs to be used.
- The `DefaultStyleKey` is missing, which is present in the `Templated Control` item template.
- The Generic.xaml paths are incorrect and should be along those lines:
  ```
  <ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
	<ResourceDictionary.MergedDictionaries>
		<ResourceDictionary Source="ms-appx:///ControlLibrary.Uno.WinUI/CustomControl.xaml" />
		<ResourceDictionary Source="ms-appx:///ControlLibrary.Uno.WinUI/Themes/CustomControlTest.xaml" />
	</ResourceDictionary.MergedDictionaries>
  </ResourceDictionary>
  ```
- The styles in `CustomControlTest.xaml` are containing an invalid `TargetType` at line 9.

We'll change the docs to make this clearer. You should also not need a shared project anymore, why did you create one?

**Comment by Soap-141** [comment]
> Thanks for the report.
> 
> 
> 
> There are multiple issues in your sample coming from misdirections in the documentation:
> 
> - "Custom Control" does not exist in the template. It's `Templated Control` that needs to be used.
> 
> - The `DefaultStyleKey` is missing, which is present in the `Templated Control` item template.
> 
> - The Generic.xaml paths are incorrect and should be along those lines:
> 
>   ```xaml
> 
>   <ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
> 
> 	<ResourceDictionary.MergedDictionaries>
> 
> 		<ResourceDictionary Source="ms-appx:///ControlLibrary.Uno.WinUI/CustomControl.xaml" />
> 
> 		<ResourceDictionary Source="ms-appx:///ControlLibrary.Uno.WinUI/Themes/CustomControlTest.xaml" />
> 
> 	</ResourceDictionary.MergedDictionaries>
> 
>   </ResourceDictionary>
> 
>   ```
> 
> - The styles in `CustomControlTest.xaml` are containing an invalid `TargetType` at line 9.
> 
> 
> 
> We'll change the docs to make this clearer. You should also not need a shared project anymore, why did you create one?

Thanks I'll take a look, and also the Shared project is used because it's a library that target both WinUI(net6) and Xamarin so we have a .Uno and .Uno.WinUI head.

**Comment by Soap-141** [comment]
@jeromelaban I updated my sample following your directive and I'm getting a Microsoft.UI.Xaml.Markup.XamlParseException: 'XAML parsing failed.'

<ResourceDictionary Source="ms-appx:///ControlLibrary.Uno.WinUI/CustomControl.xaml" />

**Comment by jeromelaban** [comment]
If you're getting this, it's because your XAML paths or resources are incorrect. Take a look at the debug output for more details.

**Comment by Soap-141** [comment]
[BringTemplate.zip](https://github.com/unoplatform/uno/files/10824854/BringTemplate.zip)

**Comment by jeromelaban** [comment]
Try breaking on all exceptions, the error message generally appears at that time.

**Comment by Soap-141** [comment]
![image](https://user-images.githubusercontent.com/57812285/221197385-c6c39d0c-1f1e-4971-beca-7441e7b0d598.png)

**Comment by jeromelaban** [comment]
If you continue running the app, what's in the output log?

**Comment by Soap-141** [comment]
![image](https://user-images.githubusercontent.com/57812285/221197572-47e5b950-88b6-4fb1-ae39-5e24ead40b6a.png)

Getting this if I continue, then the app closes.
![image](https://user-images.githubusercontent.com/57812285/221197810-6064e0da-0157-48af-b5f0-2c4ba296451a.png)

**Comment by jeromelaban** [comment]
Thanks. Another technique to debug this to enable "native debugging" and run the app again.

**Comment by Soap-141** [comment]
@jeromelaban Is there anything useful in this?? 

![Animation_72](https://user-images.githubusercontent.com/57812285/221199147-114b37e5-bf11-41d7-a1a7-c8a63f8b0639.gif)

**Comment by Soap-141** [comment]
I feel like may be the access exception could mean something?

**Comment by Soap-141** [comment]
Already tried deleting bin/obj, let me try again, and what do you mean by resources files?

**Comment by Soap-141** [comment]
Also what is this??
![image](https://user-images.githubusercontent.com/57812285/221201756-7d0c0274-c499-4b46-b07b-408fb337199b.png)

**Comment by jeromelaban** [comment]
What do you mean? You can [take a look at this document](https://platform.uno/docs/articles/uno-app-solution-structure.html) if you have questions on the app structure.

**Comment by Soap-141** [comment]
@jeromelaban I'm updating VS to 17.5 and after that I might try this https://github.com/microsoft/microsoft-ui-xaml/issues/6452

**Comment by Soap-141** [comment]
When adding <ResourceDictionary Source="ms-appx:///ControlLibrary.Uno.WinUI/Themes/Generic.xaml" /> in AppResources,xaml, I get this Cannot locate resource from 'ms-appx:///ControlLibrary.Uno.WinUI/CustomControl.xaml'. [Line: 12 Position: 25]

**Comment by Soap-141** [comment]
I don't know what I did but this file just appeared and it's working now...
![image](https://user-images.githubusercontent.com/57812285/221222993-f87107c4-6f26-45b6-8480-a3108802a23a.png)

**Comment by Soap-141** [comment]
Think it was Copy Always that was needed.

**Comment by Soap-141** [comment]
Worked when project where referenced together but not as a NuGet package.

**Comment by Soap-141** [comment]
Ok just needed to use a real NuGet build from a feed and that made Pipeline crash with a page inside the shared project but when using Xaml.Merge it was fine.
